src/renderer/processor.rs

```diff
@@ -210,6 +222,14 @@ impl<'a> Processor<'a> {
         };
 
         let len = for_loop.len();
+        if len == 0 {
+            if let Some(empty_body) = for_loop_empty_body {
+                return Ok(self.render_body(&empty_body)?);
+            } else {
+                return Ok("".to_string());
+            }
```

Comment:
I think a `match` here and on the similar if/else block above would be neater but very minor

Reply:
Good point, rewrote it using compound match statements, not sure if it's made it clearer what's going on so feel free to rework it :)

I agree with your reservations regarding the "undefined" case. Actually, couldn't you remove it and simply cover that special case with something like this:
```
{% for a in undefined_variable | default(value=[]) %}
    {{ a }}
{% else %}
    hello world
{% endfor %}
```

Reply:
It does work, the only issue is more about the principle of least surprise: an undefined condition evaluates is falsy so someone could expect an undefined contained to be falsy/empty and go into the `else` like jinja2 does.

Reply:
I can definitely see that. I guess the question boils down to how closely tera is meant to emulate Jinja2. Since it is a Rust project, I suppose could be argued that somewhat explicit typing and error-handling is preferred. Of course the end-user will likely neither know nor care about that fact.

It's your call to make, but if you land on emulating Jinja2 in this case, I think we should stick with the original naming of the empty-case block and use {% empty %}, just so there are no surprises.